src/otk/command.py

```diff
@@ -19,144 +15,182 @@
 log = logging.getLogger(__name__)
 
 
-@click.group()
-@click.option(
-    "-v",
-    "--verbose",
-    count=True,
-    help="Sets verbosity. Can be passed multiple times to be more verbose.",
-)
-@click.option(
-    "-j/",
-    "--json/--no-json",
-    default=False,
-    help="Sets output format to JSONseq. Output on stderr will be JSONseq records with ASCII record separators.",
-)
-@click.option(
-    "-i",
-    "--identifier",
-    help="An identifier to include in all log records generated by `otk -j`. Can only be used together with `-j`.",
-)
-@click.option(
-    "-w",
-    "--warn",
-    type=click.Choice(["duplicate-definition"]),
-    multiple=True,
-    help="Enable warnings, can be passed multiple times.",
-)
-@click.pass_context
-def root(
-    ctx: click.Context, verbose: int, json: bool, identifier: str, warn: list[str]
-) -> None:
-    """`otk` is the omnifest toolkit. A program to work with omnifest inputs
-    and translate them into the native formats for image build tooling."""
+def root():
+    argv = sys.argv[1:]
+
+    parser = parser_create()
 
-    ctx.ensure_object(dict)
+    if not len(argv):
+        parser.print_help()
+        parser.exit()
```

Comment:
Is this necessary?  argparse supports making an argument or subcommand a requirement which will do exactly this (print help and exit) if none are given.

Reply:
The default error provided by argparse when setting subcommands to required is more unclear to me than showing the full help:

```
(venv) user@nezhit otk € otk
usage: otk [-h] [-j] [-v] [-i IDENTIFIER] [-w WARN] {compile} ...
otk: error: the following arguments are required: command
```

I'd be fine with doing that if you feel that error is clear enough.

Reply:
Testing again, I guess we don't get the full help when things go wrong.  Not sure how much we want that vs what argparse produces (a list of valid values and a short usage line), but I think it'd be cleaner and easier to maintain if we relied on argparse functionality vs checking ourselves.

Reply:
Updated this to use built-in subcommand requirements.